File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    lowest_honer_score = [] 
    cur_score_lst = [] 
    
    for i in range(len(score)):
        cur_score_lst.append(score[i])
        if i < k:
            lowest_honer_score.append(min(cur_score_lst))
        else:
            lowest_honer_score.append(sorted(cur_score_lst, reverse=True)[k-1])
            
    return lowest_honer_score

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
const solution = (k, score) => {
    const result = [];
    const cur_score = [];
    score.forEach((i, idx) => {
        cur_score.push(i);
        if (idx < k) { result.push(Math.min(...cur_score)) }
        else {
            cur_score.sort((a, b) => b - a);
            result.push(cur_score[k-1]);
        }
    });
    return result;
}